### PR TITLE
Doc for Alpha feature PodSchedulingReadiness

### PR DESCRIPTION
**Pull request by Huang-Wei**
Part of https://github.com/kubernetes/enhancements/issues/3521

### CODE REVIEW

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -158,6 +158,7 @@ different Kubernetes components.
 | `PodDeletionCost` | `false` | Alpha | 1.21 | 1.21 |
 | `PodDeletionCost` | `true` | Beta | 1.22 | |
 | `PodHasNetworkCondition` | `false` | Alpha | 1.25 | |
+| `PodSchedulingReadiness` | `false` | Alpha | 1.26 | |
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:161]: Please add a description for this gate (at the end of this page) as we do for other features.

Comment (Huang-Wei) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:161]: will do. For now it's a placeholder.

**Comment by krol3** [comment]
Hello @Huang-Wei   👋 please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before **deadline Tuesday 15th November 2022**. Thank you!

**Comment by Huang-Wei** [comment]
@krol3 thanks for the reminder. Will get this PR ready today.

**Comment by Huang-Wei** [comment]
This PR is ready for review: https://deploy-preview-37675--kubernetes-io-vnext-staging.netlify.app/docs/concepts/scheduling-eviction/pod-scheduling-readiness/

File: content/en/docs/concepts/scheduling-eviction/pod-scheduling-readiness.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+After the Pod's creation, you can check its state using:
+
+```bash
+kubectl get pod test-pod
+```
+
+The output reveals it's in `SchedulingGated` state:
+
+```bash
```
Comment (tengqm) [inline content/en/docs/concepts/scheduling-eviction/pod-scheduling-readiness.md:60]:
````suggestion
```none
````

File: content/en/docs/concepts/scheduling-eviction/pod-scheduling-readiness.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+You can also check its `schedulingGates` field by running:
+
+```bash
+kubectl get pod test-pod -o jsonpath='{.spec.schedulingGates}'
+```
+
+The output is:
+
+```bash
```
Comment (tengqm) [inline content/en/docs/concepts/scheduling-eviction/pod-scheduling-readiness.md:73]:
````suggestion
```none
````

File: content/en/docs/concepts/scheduling-eviction/pod-scheduling-readiness.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+The output is expected to be empty. And you can check its latest status by running:
+
+```bash
+kubectl get pod test-pod -o wide
+```
+
+Given the test-pod doesn't request any CPU/memory resources, it's expected that this Pod's state get
+transited from previous `SchedulingGated` to `Running`:
+
+```bash
```
Comment (tengqm) [inline content/en/docs/concepts/scheduling-eviction/pod-scheduling-readiness.md:97]:
````suggestion
```none
````

**Review by krol3 — APPROVED** [review]
/lgtm

**Comment by sftim** [comment]
It looks like we could also update https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/ to specifically mention scheduling readiness.

https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-has-network is a similar section that already exists.

This change could land after the v1.26 release.

**Pull request merged**